### ISSUE TITLE
How to use regexp validations?

### ISSUE THREAD

**Issue by nevalla** [jamesbrooks/hash_validator#11]
I'm trying to use regexp validation with the following validation hash:

```
validations = {
  'health_check' => {
     'protocol' => /^(http|tcp)$/,
     ...
  }
}
```

When I run validation I got the following error message:

```
/Users/lauri/.rvm/gems/ruby-2.3.1/gems/hash_validator-0.6.0/lib/hash_validator/validators.rb:18:in 'validator_for': Could not find valid validator for: (?-mix:^(http|tcp)$) (StandardError)
	from /Users/lauri/.rvm/gems/ruby-2.3.1/gems/hash_validator-0.6.0/lib/hash_validator/validators/hash_validator.rb:21:in `block in validate'
	from /Users/lauri/.rvm/gems/ruby-2.3.1/gems/hash_validator-0.6.0/lib/hash_validator/validators/hash_validator.rb:20:in `each'
	from /Users/lauri/.rvm/gems/ruby-2.3.1/gems/hash_validator-0.6.0/lib/hash_validator/validators/hash_validator.rb:20:in `validate'
	from /Users/lauri/.rvm/gems/ruby-2.3.1/gems/hash_validator-0.6.0/lib/hash_validator/validators/hash_validator.rb:21:in `block in validate'
	from /Users/lauri/.rvm/gems/ruby-2.3.1/gems/hash_validator-0.6.0/lib/hash_validator/validators/hash_validator.rb:20:in `each'
	from /Users/lauri/.rvm/gems/ruby-2.3.1/gems/hash_validator-0.6.0/lib/hash_validator/validators/hash_validator.rb:20:in `validate'
	from /Users/lauri/.rvm/gems/ruby-2.3.1/gems/hash_validator-0.6.0/lib/hash_validator/base.rb:28:in `validate'
	from /Users/lauri/.rvm/gems/ruby-2.3.1/gems/hash_validator-0.6.0/lib/hash_validator/base.rb:10:in `initialize'
	from /Users/lauri/.rvm/gems/ruby-2.3.1/gems/hash_validator-0.6.0/lib/hash_validator/base.rb:19:in `new'
	from /Users/lauri/.rvm/gems/ruby-2.3.1/gems/hash_validator-0.6.0/lib/hash_validator/base.rb:19:in `validate'
	from /Users/lauri/.rvm/gems/ruby-2.3.1/gems/hash_validator-0.6.0/lib/hash_validator.rb:3:in `validate'
```

Am I doing something wrong or is this a bug?

**Comment by jamesbrooks** [comment]
Hi,

You were neither doing anything wrong, not was it actually a bug. This gem has support for validating that a regexp was present in the hash that you're validating, but not validating that a string in your validating hash conformed to a specified regexp.

I've just pushed a commit and released 0.7.0 of the gem which supports exactly what you're trying to achieve - to specify a regexp in your validations (exactly as you've specified above).

Previous this could had been accomplished using the lambda support, and called your regexp with #match within that, but I can see the appeal of having this implemented directly.

**Comment by nevalla** [comment]
Hi,

How awesome is this! I read the source code and realised the same thing and planned to go with the lambda way, but this is so great. Thanks! 🍻
